airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -376,46 +382,37 @@ private Job getJob(DSLContext ctx, long jobId) {
   }
 
   private static List<Job> getJobsFromResult(Result<Record> result) {
-    final Map<Long, List<Record>> jobIdToAttempts = result.stream().collect(Collectors.groupingBy(r -> r.getValue("job_id", Long.class)));
-
-    return jobIdToAttempts.values().stream()
-        .map(records -> {
-          final Record jobEntry = records.get(0);
-
-          List<Attempt> attempts = Collections.emptyList();
-          if (jobEntry.get("attempt_number") != null) {
-            attempts = records.stream().map(attemptRecord -> {
-              final String outputDb = attemptRecord.get("attempt_output", String.class);
-              final JobOutput output = outputDb == null ? null : Jsons.deserialize(outputDb, JobOutput.class);
-              return new Attempt(
-                  attemptRecord.get("attempt_number", Long.class),
-                  attemptRecord.get("job_id", Long.class),
-                  Path.of(attemptRecord.get("log_path", String.class)),
-                  output,
-                  Enums.toEnum(attemptRecord.get("attempt_status", String.class), AttemptStatus.class).orElseThrow(),
-                  getEpoch(attemptRecord, "attempt_created_at"),
-                  getEpoch(attemptRecord, "attempt_updated_at"),
-                  Optional.ofNullable(attemptRecord.get("attempt_ended_at"))
-                      .map(value -> getEpoch(attemptRecord, "attempt_ended_at"))
-                      .orElse(null));
-            })
-                .sorted(Comparator.comparingLong(Attempt::getId))
-                .collect(Collectors.toList());
-          }
-          final JobConfig jobConfig = Jsons.deserialize(jobEntry.get("config", String.class), JobConfig.class);
-          return new Job(
-              jobEntry.get("job_id", Long.class),
-              Enums.toEnum(jobEntry.get("config_type", String.class), ConfigType.class).orElseThrow(),
-              jobEntry.get("scope", String.class),
-              jobConfig,
-              attempts,
-              JobStatus.valueOf(jobEntry.get("job_status", String.class).toUpperCase()),
-              Optional.ofNullable(jobEntry.get("job_started_at")).map(value -> getEpoch(jobEntry, "started_at")).orElse(null),
-              getEpoch(jobEntry, "job_created_at"),
-              getEpoch(jobEntry, "job_updated_at"));
-        })
-        .sorted(Comparator.comparingLong(Job::getCreatedAtInSecond).reversed())
-        .collect(Collectors.toList());
+    List<Job> jobs = new ArrayList<Job>();
+    Job currentJob = null;
+    for (Record entry : result) {
```

Comment:
nice! much better. might be worth putting a comment to explain why we took this approach.